### PR TITLE
feat: Add the ability to request all transactions in 1 request

### PR DESCRIPTION
**Pull request by iharobCP**

### CODE REVIEW

File: packages/solana/lib/src/rpc/extension.dart [hunk 1]
```diff
@@ -32,18 +32,44 @@ extension RpcClientExt on RpcClient {
...
+  /// to the internal call to [RpcClient.getTransaction()]
+  Future<List<TransactionDetails>> getMultipleTransactions(
+    List<TransactionSignatureInformation> signatures, {
+    Commitment? commitment,
+    Encoding? encoding,
+  }) async {
+    final response = await _jsonRpcClient.bulkRequest(
+      'getTransaction',
+      signatures
+          .where((s) => s.err == null)
```
Comment (ookami-kb) [inline packages/solana/lib/src/rpc/extension.dart:57]: Why do you skip transactions with errors?

File: packages/solana/lib/src/rpc/json_rpc_client.dart [hunk 2]
```diff
@@ -15,50 +16,161 @@ class JsonRpcClient {
...
 
+    final response = await _postRequest(request);
+    if (response is _JsonRpcObjectResponse) {
+      return response.data;
+    } else {
+      throw FormatException('unexpected jsonrpc response type');
+    }
+  }
+
+  static FutureOr<Response> _onTimeout() {
```
Comment (ookami-kb) [inline packages/solana/lib/src/rpc/json_rpc_client.dart:63]: I don't see the profit of extracting this method.

File: packages/solana/lib/src/rpc/json_rpc_client.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@ import 'dart:async';
 import 'dart:convert';
 
 import 'package:http/http.dart' as http;
+import 'package:http/http.dart';
```
Comment (ookami-kb) [inline packages/solana/lib/src/rpc/json_rpc_client.dart:5]: There's already imported `http`.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/solana/lib/src/rpc/json_rpc_client.dart (hunk 3 above)
Comment (iharobCP) [reply packages/solana/lib/src/rpc/json_rpc_client.dart:5]: It should help to have the linter work this for us but great catch :D

File: packages/solana/lib/src/rpc/extension.dart (hunk 1 above)
Comment (iharobCP) [reply packages/solana/lib/src/rpc/extension.dart:57]: I havent thought this through honestly. Good catch.

File: packages/solana/lib/src/rpc/json_rpc_client.dart (hunk 2 above)
Comment (iharobCP) [reply packages/solana/lib/src/rpc/json_rpc_client.dart]: BTW: it was just to combat the very ugly formatting that resulted otherwise

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**